app/controllers/users/folders/binaries_controller.rb

```diff
@@ -0,0 +1,45 @@
+class Users::Folders::BinariesController < ApplicationController
+  before_action :set_s3_direct_post, only: [:new, :create]
+
+  def new
+    @binary = Binary.new
+  end
+
+  def create
+    user = User.find_by(username: params[:username])
+    folder = user.owned_folders.find_by(route: params[:route])
+    binary = Binary.new(binary_params)
```

Comment:
@jk1dd We're talking about adding a `current_folder` method to application controller, much like our `current_user` method